File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def setRange(newChar):
    return chr(ord('a') + ((ord(newChar) - ord('a')) % 26))

def solution(s, skip, index):
    answer = ''
    skip = set(skip)  # 효율성을 위해 set으로 변환
    
    for c in s:
        cnt = 0
        newChar = c
        
        while cnt < index:
            newChar = setRange(chr(ord(newChar) + 1))  # 다음 문자로 이동
            
            if newChar not in skip:  # skip에 없으면 count 증가
                cnt += 1
        
        answer += newChar
    
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

char setRange(char c){
    return 'a' + ((c - 'a') % 26);
}

string solution(string s, string skip, int index) {
    string answer = "";
    
    for(auto c: s){
        int cnt = 0;
        char newC = c;
        
        while(cnt != index){
            newC = setRange(newC + 1);
            
            if(skip.find(newC) == string::npos){
                cnt++;
            }
        }
        answer += newC;
    }
    return answer;
}